src/__tests__/middleware.test.ts

```diff
@@ -82,4 +152,40 @@ describe('Response mutation', () => {
     const result = await lambdaHandler(event, {} as any, () => {})!;
     expect(result.cookies).toContain(cookieValue);
   });
+  it('is allowed to access updated context in result middleware', async () => {
+    const event = createEvent(gql`
+      mutation {
+        mutateContext
+      }
+    `);
+    const lambdaHandler = startServerAndCreateLambdaHandler<
+      handlers.RequestHandler<
+        APIGatewayProxyEventV2,
+        APIGatewayProxyStructuredResultV2
+      >,
+      {
+        foo: string | null;
+      }
+    >(server, handlers.createAPIGatewayProxyEventV2RequestHandler(), {
+      context: async () => {
+        return {
+          foo: 'bar',
```

Comment:
This test doesn't actually show the context being updated, should it?
It starts as `foo: 'bar'` and the mutation updates it to `foo: 'bar'`.

```suggestion
          foo: '',
```

Reply:
This change doesn't work as I expected, so I'm misunderstanding I think. Shouldn't `foo` be updated to `bar` by the resolver?